schemas/constructs/core.json

```diff
@@ -36,6 +31,406 @@
+        },
+        "color": {
+          "type": "string",
+          "description": "The color of the element's label. Colours may be specified by name (e.g. red), hex (e.g. #ff0000 or #f00), RGB (e.g. rgb(255, 0, 0)), or HSL (e.g. hsl(0, 100%, 50%))."
+        },
+        "text-opacity": {
+          "type": "number",
+          "description": "The opacity of the label text, including its outline.",
+          "minimum": 0,
+          "maximum": 1
+        },
+        "font-family": {
+          "type": "string",
+          "description": "A comma-separated list of font names to use on the label text."
+        },
+        "font-size": {
+          "type": "string",
+          "description": "The size of the label text."
+        },
+        "font-style": {
+          "type": "string",
+          "description": "A CSS font style to be applied to the label text."
+        },
+        "font-weight": {
+          "type": "string",
+          "description": "A CSS font weight to be applied to the label text."
+        },
+        "text-transform": {
+          "type": "string",
+          "description": "A transformation to apply to the label text",
+          "enum": ["none", "uppercase", "lowercase"]
+        },
+        "opacity": {
+          "type": "number",
+          "description": "The opacity of the element. See https://js.cytoscape.org/#style/visibility",
+          "minimum": 0,
+          "maximum": 1
+        },
+        "z-index": {
+          "type": "integer",
+          "description": "An integer value that affects the relative draw order of elements. In general, an element with a higher z-index will be drawn on top of an element with a lower z-index. Note that edges are under nodes despite z-index."
+        },
+        "label": {
+          "type": "string",
+          "description": "The text to display for an element\u2019s label. Can give a path, e.g. data(id) will label with the elements id"
+        }
+      }
+    },
+    "shape": {
+      "type": "string",
+      "description": "The shape of the node\u2019s body. Note that each shape fits within the specified width and height, and so you may have to adjust width and height if you desire an equilateral shape (i.e. width !== height for several equilateral shapes)",
+      "enum": [
+        "ellipse",
+        "triangle",
+        "round-triangle",
+        "rectangle",
+        "round-rectangle",
+        "bottom-round-rectangle",
+        "cut-rectangle",
+        "barrel",
+        "rhomboid",
+        "diamond",
+        "round-diamond",
+        "pentagon",
+        "round-pentagon",
+        "hexagon",
+        "round-hexagon",
+        "concave-hexagon",
+        "heptagon",
+        "round-heptagon",
+        "octagon",
+        "round-octagon",
+        "star",
+        "tag",
+        "round-tag",
+        "vee",
+        "polygon"
+      ]
+    },
+
+    "relationshipStyles": {
```

Comment:
Relationships, not edges? @codeSafari10?

These styles apply to hierarchical, sibling, and other non-edge-based relationships?

Reply:
the generic set of styles apply to all , i will split the ones specif to edge